Updating glue_job documentation to provide example for Glue Job v5.0

trutled3 opened the pull request:
<!---
See what makes a good Pull Request at: https://hashicorp.github.io/terraform-provider-aws/raising-a-pull-request/
--->
### Description
<!---
Please provide a helpful description of what change this pull request will introduce.
--->
* This pull request updates the documentation for the `aws_glue_job` resource to notify the consumer that setting `python_version` to `3` when setting `glue_version` to `"5.0"` uses Python v3.11 instead of v3.6 as the the documentation previously stated.


### Relations
<!---
If your pull request fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates.

For Example:

Relates #0000
or 
Closes #0000
--->

Closes #41213 

### References
<!---
Optionally, provide any helpful references that may help the reviewer(s).
--->
* N/A

justinretzolk reviewed (changes requested):
Hey @trutled3 👋 Thank you for taking the time to raise this! We don't make changelog entries for documentation updates, so `.changelog/41636.txt` should be removed before we can get this merged. Once you get that done and the checks run again, I'd be happy to get this merged 🙂

trutled3 commented:
My mistake, thank you!  I will get that removed.

trutled3 commented:
@justinretzolk - Thank you again for reviewing my pull request! I have removed the changelog file and corrected the formatting error from `terrafmt`. Please review at your earliest convenience.

justinretzolk reviewed (approved):
Looks good to me! Thanks for getting those updates in @trutled3, and thanks again for the contribution! 🎉
